src/components/Envelope.vue

```diff
@@ -85,39 +85,62 @@
 				</ActionButton>
 			</EnvelopePrimaryActions>
 			<template v-if="!moreActionsOpen">
-				<ActionButton icon="icon-junk"
-					:close-after-click="true"
+				<ActionButton :close-after-click="true"
 					@click.prevent="onToggleJunk">
+					<template #icon>
+						<AlertOctagonIcon
```

Comment:
this doesnt work, when i mark as spam, the icon is not shown

Reply:
The action icon  renders for spam and non-spam message. Clicking toggles the state. Then I can toggle back. This works in the envelope list and in the message view.

What exactly does not show for you?

Reply:
for me works, besides icons inside of EnvelopePrimaryActions

Reply:
Sorry, i was testing this wrong, as i said on my first comment, when i toggle mark as span, the icon doesnt show on the avatar (not related to this pr). 
What you have done here is changing the left icon on the menu.